### PR TITLE
Fix for specifying handler with period in path

### LINKED ISSUE #583: Function: cannot specify handler with dot in the path
Currently, defining the handler path like this:
```
handler: `./src/lambda.handler`,
```
causes a build error:
```
Transpiling Lambda code...
 > error: Could not resolve "../.js"
```

---
Notify ppl who ran into this:
- Dax (Slack thread https://serverless-stack.slack.com/archives/C01HQQVC8TH/p1626895089465300)
- Brent (GH issue https://github.com/serverless-stack/serverless-stack/issues/589)

### PR DESCRIPTION
**Pull request by thdxr**
Fixes #583

### CODE REVIEW

File: packages/cli/scripts/start.js [hunk 1]
```diff
@@ -476,14 +476,10 @@ async function handleRemoveWatchedFiles(files) {
...
 
-async function handleTranspileNode({
-  srcPath,
-  handler,
-  bundle,
-  esbuilder,
-  onSuccess,
-  onFailure,
-}, config) {
+async function handleTranspileNode(
```
Comment (thdxr) [inline packages/cli/scripts/start.js:479]: Seems like my prettier is formatting differently but I am on latest

File: packages/cli/scripts/start.js [hunk 2]
```diff
@@ -693,18 +689,19 @@ function handleRunTypeCheck(srcPath, inputFiles, tsconfig, config) {
...
   for (const ext of extensions) {
-    const file = path.join(appPath, srcPath, `${name}${ext}`);
+    const file = path.join(appPath, srcPath, dir, `${name}${ext}`);
     if (await checkFileExists(file)) {
       return file;
     }
   }
 
-  return path.join(appPath, srcPath, `${name}.js`);
+  return path.join(appPath, srcPath, dir, `${name}.js`);
```
Comment (thdxr) [inline packages/cli/scripts/start.js:704]: Curious why it defaults to this without verifying it exists

Comment (fwang) [reply packages/cli/scripts/start.js:704]: Yeah we should. Let me add a check.

File: packages/cli/scripts/start.js [hunk 3]
```diff
@@ -693,18 +689,19 @@ function handleRunTypeCheck(srcPath, inputFiles, tsconfig, config) {
 }
 
 async function getHandlerFilePath(appPath, srcPath, handler) {
-  const parts = handler.split(".");
+  const parts = path.basename(handler).split(".");
```
Comment (fwang) [inline packages/cli/scripts/start.js:692]: It doesn't handle this case, as suggested here - https://github.com/serverless-stack/serverless-stack/issues/600
```
handler: 'src/users/index.controller.main'
```

We actually do something similar on `sst deploy`. Let's re-use the same logic - https://github.com/serverless-stack/serverless-stack/blob/master/packages/resources/src/util/nodeBuilder.ts#L130

**Comment by fwang** [comment]
Hey @thdxr, I'm not able to push changes to your branch. So I created forked your branch and created a new PR https://github.com/serverless-stack/serverless-stack/pull/611

**Pull request merged**